core/src/main/java/com/graphhopper/routing/util/spatialrules/SpatialRuleLookupArray.java

```diff
@@ -105,10 +115,10 @@ public SpatialRule lookupRule(double lat, double lon) {
             }
         }
 
-        return EMPTY_RULE;
+        return EMPTY;
     }
 
-    protected int getRuleIndex(int xIndex, int yIndex) {
+    protected int getRuleContainerIndex(int xIndex, int yIndex) {
         if (xIndex < 0 || xIndex > lookupArray.length) {
```

Comment:
Well, usually ifs for boundary checks are very fast. For now I wouldn't care that much. But we could check the result (in lookupRule) for EMPTY_RULE_INDEX and return EMPTY instead of the call to ruleContainers etc?